content/blog/2016/2016-12-13-jenkins-puppet-enterprise-plugin.adoc

```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: "Continuous delivery with Jenkins and Puppet Enterprise"
+tags:
+- jenkins
+- continuousdelivery
+- puppet
+author: ccaum
+---
+
+During PuppetConf 2016, myself and Brian Dawson from CloudBees announced the link:https://wiki.jenkins-ci.org/display/JENKINS/Puppet+Enterprise+Pipeline+Plugin[Puppet Enterprise plugin for Jenkins Pipeline.] Let's take a look at how the plugin makes it trivial to use Puppet to perform some or all of the deployment tasks in continuous delivery pipelines.
+
+Jenkins Pipeline introduced an amazing world where pipelines are defined and managed from the version control repository that houses the code the pipeline delivers. This is a powerful idea, and one I fealt complemented Puppet's automation strengths. I wanted to make it trivial to control Puppet Enterprise's orchestration and infrastructure code management capabilities, as well as control Hiera data and use PuppetDB as a source of truth – all from a Pipeline script. The result was a plugin that fully buys into the Pipeline ideals by providing methods to control the different capabilities in Puppet Enterprise. The methods provide ways to query PuppetDB, set Hiera key/value pairs, deploy Puppet code environments, and kick off orchestrated Puppet runs.
```

Comment:
The result was a plugin that -> The result was the Puppet Enterprise for Jenkins Pipeline plugin, which